### PR TITLE
Fix favorite tag overflow and add h1&h2 headers 

### PR DESCRIPTION
**Pull request by gracez72**
Tickets: [TASK-34465](https://microsoftgarage.visualstudio.com/Intern%20GitHub/_workitems/edit/34465), [BUG-34620](https://microsoftgarage.visualstudio.com/Intern%20GitHub/_workitems/edit/34620)

Favorite tags that overflow now cut off and show ellipses. 
overflowtags

This PR also updates divs to h1 or h2 tags based on accessibility review.

### CODE REVIEW

File: private-templates-service/client/src/components/NavBar/styled.tsx [hunk 1]
```diff
@@ -59,7 +59,7 @@ export const Styledh1 = styled.div`
   }
 `;
 
-export const Styledh2 = styled.div`
+export const Styledh2 = styled.h2`
```
Comment (daniellemott) [inline private-templates-service/client/src/components/NavBar/styled.tsx:62]: Can you please confirm that changing this to h2 didn't make something that was not bold become bold?

Comment (gracez72) [reply private-templates-service/client/src/components/NavBar/styled.tsx:62]: Yep, nothing changed to bold!

**Comment by beervoley** [comment]
Could you please confirm that display of tags on All Cards page is still the same and was not affected?

**Comment by gracez72** [comment]
@beervoley All Cards tags are not affected!

**Pull request merged**